sorting/dnf_sort.cpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file
+ * @brief Implementation of DNF
+ * sort(https://www.geeksforgeeks.org/sort-an-array-of-0s-1s-and-2s/)
+ * @details
+ * C++ program to sort an array with 0, 1 and 2 in a single pass(DNF sort).
+ * Since one traversal of the array is there hence it works in O(n) time
+ * complexity.
+ * @author [GeeksForGeeks](https://www.geeksforgeeks.org/)
+ */
+
+#include <algorithm>  /// for std::is_sorted
+#include <cassert>    /// for assert
+#include <iostream>   /// for std::swap and io operations
+#include <vector>     /// for std::vector
+/**
+ * @namespace sorting
+ * @breif Sorting algorithms
+ */
+namespace sorting {
+/**
+ * @namespace dnf_sort
+ * @brief Functions for [DNF
+ * sort](https://en.wikipedia.org/wiki/Dutch_national_flag_problem)
+ */
+namespace dnf_sort {
+/**
+ * @brief The main function implements DNF sort
+ * @tparam T type of array
+ * @param a array to be sorted,
+ * @param arr_size size of array
+ * @returns void
+ */
+template <typename T>
+std::vector<T> dnfSort(const std::vector<T> &in_arr) {
+    std::vector<T> arr(in_arr);
+    int lo = 0;
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂

Reply:
I think you are referring to these 3 variables being used `lo` , `hi` and `mid`  to make them `uint64_t` instead of `int`.

Reply:
Correct.